### ISSUE TITLE
Render (up)scaling

### ISSUE THREAD

**Issue by lukirudaka** [xCollateral/VulkanMod#503]
### Describe the request

Add the ability to render in, for example, 1080p and upscale to 4k.

This will be useful for those wishing to use more eye-candy settings on less-powerful GPUs. In my case: Increase render distance on the Adreno 618 from 5 chunks@20FPS 2000x1200 to.. maybe 8 chunks @ 30 FPS at a resolution of 1000x600 (half window resolution upscaled to 2000x1200)

There are mods that would do this such as ResolutionControl3 (and the mods it was forked from), but those mods are broken with VulkanMod (for obvious reasons) (and the Adreno 618's desktop OGL drivers are... iffy at best).

**Comment by lukirudaka** [comment]
("Isn't the Adreno 618 for Android devices?" Well yes, but actually no, I'm running Ubuntu :D)

**Comment by Felix14-v2** [comment]
I don't think that the fillrate of your GPU is a bottleneck if we speak about increasing the render distance. To easily handle more chunk sections, you have to implement faster invisible chunk culling algorithms (frustum and occlusion culling). This is a CPU-limited scenario, not GPU. But if you want a lower rendering resolution, you can just reduce the fullscreen resolution option.

Anyways, https://github.com/xCollateral/VulkanMod/issues/136

**Comment by Andarwinux** [comment]
This is also useful for more powerful desktop GPUs, I rendering UI at 6K via DLDSR on a 4K display due to Windows' shitty font rendering. RTX4070 can certainly rendering Minecraft at 6K without any problems, but it's not a substantial improvement like font rendering, I just want it to continue rendering at 4K to keep the fans down.
